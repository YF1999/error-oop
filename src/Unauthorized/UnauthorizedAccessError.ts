import { IErrorArguments, IErrorOptions, NativeError } from '../Native';
import { ErrorTool } from '../Tools';

export interface IUnauthorizedAccessErrorOptions extends IErrorOptions {}

/**
 * Applicable when the operating system denies access because of an I/O error or a specific type of security error.
 */
export class UnauthorizedAccessError extends NativeError {
    public constructor();
    /**
     * @param message The error message that explains the reason for this error.
     */
    public constructor(message: string);
    /**
     * @param message The error message that explains the reason for this error.
     * @param innerError The error that is the cause of the current error. Stack trace will be appended.
     */
    public constructor(message: string, innerError: Error);
    /**
     * @param options The constructor options.
     */
    public constructor(options: IUnauthorizedAccessErrorOptions);

    public constructor(...args: IErrorArguments<IUnauthorizedAccessErrorOptions>) {
        super(ErrorTool.parseErrorArguments([], ...args));
    }
}
