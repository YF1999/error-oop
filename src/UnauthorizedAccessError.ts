import { ErrorOptions, UnauthorizedAccessErrorMessageProps, UnauthorizedAccessErrorProps } from './CommonTypes';
import { AbstractError } from './Native';

export abstract class AbstractUnauthorizedAccessError extends AbstractError {
    public constructor(
        props: UnauthorizedAccessErrorProps,
        options: ErrorOptions<UnauthorizedAccessErrorMessageProps>,
    ) {
        super(props, options);
    }
}

/**
 * Applicable when the operating system denies access because of an I/O error or a specific type of
 * security error.
 */
export class UnauthorizedAccessError extends AbstractUnauthorizedAccessError {
    public constructor();
    /**
     * @param message The error message that explains the reason for this error.
     */
    public constructor(message: string);
    /**
     * @param message The error message that explains the reason for this error.
     * @param innerError The error that is the cause of the current error. Stack trace will be append.
     */
    public constructor(message: string, innerError: Error);

    public constructor(message: string = '', innerError?: Error) {
        super({ message, innerError }, {});
    }
}
