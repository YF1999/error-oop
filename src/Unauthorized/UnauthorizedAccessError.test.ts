import { UnauthorizedAccessError } from './UnauthorizedAccessError';

standardTestError(UnauthorizedAccessError, 'UnauthorizedAccessError', 'error');
