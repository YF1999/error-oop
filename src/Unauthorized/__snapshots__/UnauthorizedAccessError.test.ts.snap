// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test \`UnauthorizedAccessError\` own keys 1`] = `
Array [
  "stack",
]
`;

exports[`Test \`UnauthorizedAccessError\` own keys 2`] = `
Array [
  "stack",
  "message",
]
`;

exports[`Test \`UnauthorizedAccessError\` own keys 3`] = `
Array [
  "stack",
  "message",
]
`;

exports[`Test \`UnauthorizedAccessError\` own keys 4`] = `
Array [
  "stack",
  "message",
]
`;
